appland/popup/record.js

```diff
@@ -1,23 +1,49 @@
+const chrome = (window.chrome || window.browser);
 const statusElement = document.querySelector('.status');
 const recordButton = document.querySelector('#appmap-record');
 const recordButtonGraphic = document.querySelector('.appmap-record-button');
 const urlInput = document.querySelector("#appland-url");
 const header = document.querySelector('.header');
-const browser = (window.chrome || window.browser);
 
 const colorReady = '#da0030';
 const colorDisabled = '#350020';
 
 let ellipsisTimeoutId = -1;
 
+document.addEventListener('DOMContentLoaded', onLoad, false);
+header.addEventListener('click', (e) => {
+  const url = getAppLandUrl();
+  if (!isUrlValid(url)) {
+    return;
+  }
+
+  window.open(url, '_blank');
+});
+
+recordButton.addEventListener('change', (e) => {
+  e.target.checked ? startRecording() : stopRecording()
+});
+
+urlInput.addEventListener('change', (e) => {
+  saveApplandUrl(e.target.value);
+});
+
+
+function showXHRError(req, msg) {
+  console.log(msg);
+  console.log(`status: ${req.status}`);
+  console.log(req.response);
+  alert(msg);
+}
+  
 function saveApplandUrl(url) {
-  browser.storage.local.set({applandUrl: url});
+  chrome.storage.local.set({applandUrl: url});
 }
 
 function getTabUrl(callback) {
-  browser.tabs.query({active: true, currentWindow: true}, (tabs) => {
+  chrome.tabs.query({active: true, currentWindow: true}, (tabs) => {
     const url = new URL(tabs[0].url);
-    callback(url);
+    callback(url, tabs[0].windowId);
   });
 }
```

Comment:
Is it possible to merge the other PR into this branch? I didn't realize the PR was still outstanding. It allows us to specify an alternate hostname to record from if the backend uses a different hostname.